viscy/translation/evaluation.py

```diff
@@ -12,43 +15,254 @@
-                "mAP": coco_metrics["map"],
-                "mAP_50": coco_metrics["map_50"],
-                "mAP_75": coco_metrics["map_75"],
-                "mAR_100": coco_metrics["mar_100"],
-            }
+
+        # Create metrics dictionary
+        metrics_dict = {
+            "position": batch["position_idx"][0],
+            "time": batch["time_idx"][0],
+            "accuracy": (accuracy(pred_binary, target_binary, task="binary")),
+            "dice": (
+                dice_score(
+                    pred_binary.long()[None],
+                    target_binary.long()[None],
+                    num_classes=2,
+                    input_format="index",
+                )
+            ),
+            "jaccard": (jaccard_index(pred_binary, target_binary, task="binary")),
+            "mAP": coco_metrics["map"],
+            "mAP_50": coco_metrics["map_50"],
+            "mAP_75": coco_metrics["map_75"],
+            "mAR_100": coco_metrics["mar_100"],
+        }
+
+        # Add metadata if available
+        if "cell_type" in batch:
+            metrics_dict["cell_type"] = batch["cell_type"]
+        if "organelle" in batch:
+            metrics_dict["organelle"] = batch["organelle"]
+        if "infection_condition" in batch:
+            metrics_dict["infection_condition"] = batch["infection_condition"]
+
+        self.logger.log_metrics(metrics_dict)
+
+    def _compute_3d_metrics(self, pred, target, pred_binary, target_binary, batch):
+        """Compute and log metrics for 3D segmentation."""
+        unique_instances_target = np.unique(target)
+        unique_instances_pred = np.unique(pred)
+
+        _logger.debug(
+            f"Unique instances: {unique_instances_target} and {unique_instances_pred}"
         )
+
+        ## Measuring Panoptic Quality
+        evaluator = Panoptica_Evaluator(
+            expected_input=InputType.UnmatchedInstancePair,
+            instance_metrics=[Metric.DSC, Metric.IoU],
+            decision_metric=Metric.DSC,
+            decision_threshold=0.5,
+            log_times=True,
+        )
+        result = evaluator.evaluate(pred, target, verbose=False)
+        result = result.to_dict()
+        _logger.debug(result)
+
+        # Create metrics dictionary
+        metrics_dict = {
+            "position": batch["position_idx"][0],
+            "time": batch["time_idx"][0],
+            "target_instances": unique_instances_target,
+            "pred_instances": unique_instances_pred,
+            **result,
+        }
+
+        # Add metadata if available
+        if "cell_type" in batch:
+            metrics_dict["cell_type"] = batch["cell_type"]
+        if "organelle" in batch:
+            metrics_dict["organelle"] = batch["organelle"]
+        if "infection_condition" in batch:
+            metrics_dict["infection_condition"] = batch["infection_condition"]
+
+        self.logger.log_metrics(metrics_dict)
```

Comment:
This is repeated in 2D and 3D. Should return the metrics (as the 'compute' in the method name would suggests) and let the calling function log once.